test/unittest/tests/wrappers/connection_wrapper.cpp

```diff
@@ -5,26 +5,46 @@
  *
  * See the file LICENSE for redistribution information.
  */
+#include <cstdio>
+
+#include "wiredtiger.h"
+#include "wt_internal.h"
 
 #include "connection_wrapper.h"
-#include "error_handler.h"
+#include "../utils.h"
 
-ConnectionWrapper::ConnectionWrapper(WT_CONNECTION_IMPL *connectionImpl)
-    : _connectionImpl(connectionImpl)
+ConnectionWrapper::ConnectionWrapper() : _conn_impl(nullptr), _conn(nullptr)
 {
+    utils::throwIfNonZero(wiredtiger_open(nullptr, nullptr, "create", &_conn));
 }
 
 ConnectionWrapper::~ConnectionWrapper()
 {
-    __wt_free(nullptr, _connectionImpl);
+    utils::throwIfNonZero(_conn->close(_conn, ""));
```

Comment:
Destructors should not emit exceptions (see Effective C++, 3rd ed, item 8, pages 44+) as it causes all sorts of issues (usually terminating the program)
In this case, I suggest we just close the connection (removing the throwIfNonZero() call) and log an error if close fails.

Reply:
Ok, let's leave that unchanged for now.